src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2354,13 +2458,38 @@ public void HandleIDebugThreadDestroyEvent2(IDebugEngine2 pEngine, IDebugProcess
         }
 
         public void HandleIDebugModuleLoadEvent2(IDebugEngine2 pEngine, IDebugProcess2 pProcess, IDebugProgram2 pProgram, IDebugThread2 pThread, IDebugEvent2 pEvent)
+        {
+            IDebugModule2 module;
+            string moduleLoadMessage = null;
+            int isLoad = 0;
+            ((IDebugModuleLoadEvent2)pEvent).GetModule(out module, ref moduleLoadMessage, ref isLoad);
+            
+            m_logger.WriteLine(LoggingCategory.Module, moduleLoadMessage);
+
+            var debugModuleInfos = new MODULE_INFO[1];
+            if (module.GetInfo(enum_MODULE_INFO_FIELDS.MIF_ALLFIELDS, debugModuleInfos) == HRConstants.S_OK)
+            {
+                var mod = ConvertToModule(debugModuleInfos[0], module);
+                Protocol.SendEvent(new ModuleEvent(ModuleEvent.ReasonValue.New, mod));
+            }
+        }
+
+        public void HandleIDebugModuleUnloadEvent2(IDebugEngine2 pEngine, IDebugProcess2 pProcess, IDebugProgram2 pProgram, IDebugThread2 pThread, IDebugEvent2 pEvent)
```

Comment:
This callback never happens since a module unload is a `IDebugModuleLoad2Event` with `isLoad = false`.

You want to move this block:
```
            var debugModuleInfos = new MODULE_INFO[1];
            if (module.GetInfo(enum_MODULE_INFO_FIELDS.MIF_ALLFIELDS, debugModuleInfos) == HRConstants.S_OK)
            {
                var mod = ConvertToModule(debugModuleInfos[0], module);
                ReleaseDebugModuleId((IntPtr)mod.Id);
                Protocol.SendEvent(new ModuleEvent(ModuleEvent.ReasonValue.Removed, mod));
            }
```
into `HandleIDebugModuleLoadEvent2` when isLoad is `false`.

Reply:
This is where the call back is registered.
https://github.com/microsoft/MIEngine/blob/61702cd8f53d20dedd8c694a6fc00a51b0ad3f17/src/OpenDebugAD7/AD7DebugSession.cs#L2252

There is no interface for `IDebugModuleUnloadEvent2`.